src/accountsdb/db.zig

```diff
@@ -2030,6 +2004,70 @@ pub const AccountsDB = struct {
         return biggest;
     }
 
+    /// TODO: a number of these parameters are temporary stand-ins for data that will be derived from the state
+    /// of AccountsDB, which currently doesn't all exist.
+    pub fn writeSnapshotTarTo(
+        /// Although this is a mutable pointer, this method performs no mutations;
+        /// the mutable reference is simply needed in order to obtain a lock on some
+        /// fields.
+        self: *Self,
+        /// `std.io.GenericWriter(...)` | `std.io.AnyWriter`
+        archive_writer: anytype,
+        status_cache: StatusCache,
+        bank_fields: sig.accounts_db.snapshots.BankFields,
+        bank_fields_inc: sig.accounts_db.snapshots.BankFields.Incremental,
+        lamports_per_signature: u64,
+        bank_hash_info: sig.accounts_db.snapshots.BankHashInfo,
+        /// For tests against older snapshots
+        stored_meta_write_version: u64,
+    ) !void {
+        const file_map, var file_map_lg = self.file_map.readWithLock();
+        defer file_map_lg.unlock();
+
+        const max_rooted_slot = self.largest_root_slot.load(.unordered);
+        const version = sig.version.CURRENT_CLIENT_VERSION;
+
+        const account_file_infos_buf = try self.allocator.alloc(AccountFileInfo, file_map.count());
+        defer self.allocator.free(account_file_infos_buf);
+
+        var serializable_file_map = std.AutoArrayHashMap(Slot, []const AccountFileInfo).init(self.allocator);
+        defer serializable_file_map.deinit();
+        try serializable_file_map.ensureTotalCapacity(file_map.count());
+
+        for (file_map.values(), account_file_infos_buf) |*value_guarded, *info| {
+            const acc_file, var value_lg = value_guarded.readWithLock();
+            defer value_lg.unlock();
+
+            if (acc_file.slot > max_rooted_slot) continue;
+
+            info.* = .{
+                .id = acc_file.id,
+                .length = acc_file.length,
+            };
+
+            serializable_file_map.putAssumeCapacityNoClobber(acc_file.slot, info[0..1]);
+        }
+
+        const snapshot_fields: SnapshotFields = .{
+            .bank_fields = bank_fields,
+            .accounts_db_fields = .{
+                .file_map = serializable_file_map,
+
+                .stored_meta_write_version = stored_meta_write_version,
+
+                .slot = max_rooted_slot,
+                .bank_hash_info = bank_hash_info,
+
+                .rooted_slots = .{},
+                .rooted_slot_hashes = .{},
+            },
+            .lamports_per_signature = lamports_per_signature,
+            .bank_fields_inc = bank_fields_inc,
+        };
+
```

Comment:
We have access to `@import("zstd").Writer`, which we can pipe this into, we just don't do that anywhere yet.

Reply:
Well the thing is the functionality is already "exposed", you just wrap a writer in the zstd writer, and then pass that to this function, and voila, compression. There just isn't anywhere where this makes sense to do. I can go ahead and throw an example in a demonstrative test.

Reply:
can you add an other method 'writeSnapshot' which uses the zstd decompression writer by default (probs should be in another pr) - also probs worth thinking about efficient ways to do this - since this blocks flush/clean/... - ideally we can start compression after an account file has been added to the tar archive buffer - would likely need multiple threads ...

Reply:
Yeah, that was vaguely how I was picturing it, just like I said, this doesn't slot in anywhere yet so it's hard to judge the best abstraction for it, the main thing I've focused on here is building the infrastructure to make it easy enough to integrate when the time comes.